### PR TITLE
Add 'hide' property to Pokemon

### PR DESCRIPTION
**Pull request by KrisXV**
This will mostly be used for the smogdex and for the teambuilder tables.

### CODE REVIEW

File: sim/dex-data.ts [hunk 1]
```diff
@@ -606,7 +606,9 @@ export class Template extends BasicEffect implements Readonly<BasicEffect & Temp
 	 */
 	readonly learnset?: {[moveid: string]: MoveSource[]};
 	/** Source of learnsets for Pokemon that lack their own */
-	readonly inheritsFrom: string | string[];
+	readonly inheritsFrom?: string | string[];
+	/** Whether or not a Pokemon should be hidden in the teambuilder */
+	readonly hide?: 'SpriteOnly' | 'SpriteMinute' | 'Minute';
```
Comment (Zarel) [inline sim/dex-data.ts]: Neither the word `hide` nor these possible values are very clear to me.

`SpriteOnly` appears to be a cosmetic form? I don't understand `SpriteMinute` and `Minute` at all.

File: data/pokedex.js [hunk 2]
```diff
@@ -430,6 +432,7 @@ let BattlePokedex = {
 		weightkg: 105,
 		color: "Black",
 		eggGroups: ["Field"],
+		hide: "SpriteOnly",
```
Comment (monsanto) [inline data/pokedex.js]: I believe @Marty-D said the totems do have competitive differences beyond their sprite, such as weight (which matters for doubles?)

File: data/pokedex.js [hunk 3]
```diff
@@ -9346,6 +9376,7 @@ let BattlePokedex = {
 		evoLevel: 35,
 		eggGroups: ["Field"],
 		inheritsFrom: "darmanitan",
+		hide: "SpriteMinute",
```
Comment (monsanto) [inline data/pokedex.js]: Not sure what property you're going here (further elaboration from myself and TDP on Discord)

File: data/pokedex.js [hunk 4]
```diff
@@ -10749,6 +10784,7 @@ let BattlePokedex = {
 		color: "Purple",
 		eggGroups: ["Undiscovered"],
 		inheritsFrom: "genesect",
+		hide: "Minute",
```
Comment (monsanto) [inline data/pokedex.js]: What differences does this have beyond sprite?

File: data/pokedex.js [hunk 5]
```diff
@@ -12451,6 +12497,7 @@ let BattlePokedex = {
 		weightkg: 78.6,
 		color: "Blue",
 		eggGroups: ["Water 2"],
+		hide: "Minute",
```
Comment (monsanto) [inline data/pokedex.js]: I would argue there is a huge difference here

File: data/pokedex.js [hunk 6]
```diff
@@ -14848,6 +14921,7 @@ let BattlePokedex = {
 		weightkg: 89,
 		color: "Blue",
 		eggGroups: ["Water 1", "Field"],
+		hide: "Minute",
```
Comment (monsanto) [inline data/pokedex.js]: Looks big to me

File: sim/dex-data.ts (hunk 1 above)
Comment (monsanto) [reply sim/dex-data.ts]: `SpriteOnly` refers to formes that only differ from their base in appearance. The difference with cosmetic formes is that their appearance can convey competitively-relevant information. For example if you see Mimikyu-Busted you know that Disguise is no longer in play.

`Minute` (as in, the adjective meaning small/insignificant, and not the unit of time) refers to formes that do technically have distinctions from their base, but not very many. It makes sense to tier these mons together. Any strategy we write would cover both of them. We'd display them on the same Dex page, noting the few differences. Basculin-Blue-Striped is a good example.

Per Discord discussion I'm not sure what `SpriteMinute` is supposed to be.

**Comment by monsanto** [comment]
I'd call this property something like `differences` rather than `hide`.

File: data/pokedex.js (hunk 2 above)
Comment (KrisXV) [reply data/pokedex.js]: i dont think any of the totems are used competitively in smogon formats at least (except for a small period in time where totem araquanid was viable in RU)

File: sim/dex-data.ts (hunk 1 above)
Comment (Zarel) [reply sim/dex-data.ts]: I don't like `Minute`, maybe something more like `Minor`.

I also don't like `hide`, I think I'd prefer `unimportantForme`.

Comment (Zarel) [reply sim/dex-data.ts]: I also want the "can convey competitively-relevant information" clarification in the doc comment.

**Pull request closed without merging**